src/Drivers/Database.php

```diff
@@ -23,10 +23,26 @@ public function audit(Auditable $model): ?Audit
     public function prune(Auditable $model): bool
     {
         if (($threshold = $model->getAuditThreshold()) > 0) {
-            return $model->audits()
+            $class = get_class($model->audits()->getModel());
+            $auditKeyName = (new $class)->getKeyName();
+
+            $forRemoval = $model->audits()
                 ->latest()
-                ->offset($threshold)->limit(PHP_INT_MAX)
-                ->delete() > 0;
+                ->get()
+                ->slice($threshold)
+                ->pluck($auditKeyName);
+
+            if (!$forRemoval->isEmpty()) {
+                $forRemovalChunks = $forRemoval->chunk(Config::get('audit.placeholders_limit', 50000));
```

Comment:
If you have thousands of records, they will all be stored in memory as models, wouldn't that be less efficient than a `leftJoin`?